### ISSUE TITLE
Hey guys why don't you just have done a PR to my CMS

### ISSUE THREAD

**Issue by batiste** [ArabellaTech/django-basic-cms#11]
Hi people from ArabellaTech,

I just discovered this repository.

First thanks for at least giving me credits but apart from just changing the name of the project what prevented you to contribute to the original project?

I would have greatly appreciated any contributions to the orignal repository and I am sure we could have found a common ground for development.

Do you still use this code base? Is there anything in there I should try to backport to the original project?

**Comment by remik** [comment]
Hi,
Your project was dead for sometime (we did some PR and it was waiting for a long time). 
Our main changes are to support SEO by default + small fixes.
We also plan to do new import/export (JSON + media files via website (oAuth connection) - not import/export).
@jacoor any hint what we can backport or can we sync to main project?

**Comment by jacoor** [comment]
Hello
I don't think we need to backport anything - stuff right now works for us. But, according to http://pythonhosted.org/django-page-cms/introduction.html#dependencies-compatibility original project does not support django > 1.7, which we did in version > 3.0 (and actually left earlier versions support at 0.2.x)
I'd go with merge if possible.

**Comment by batiste** [comment]
Sorry but the documentation is not fully up to date.
Right now the original project is compatible with Django 1.8, I think we really duplicated the work on maintaining and pushing it to 1.8.

I would love a merge if possible. But I imagine the fact you guys changed the name is an impediment. I planned to change the name of the module to gerbi (Gerbi CMS) because it seemed like a more appealing name. I decided against it after realising the mess that will be created in with database migrations.

Tell me if changing the name of the module to page is an issue for you guys because it is obviously the  elephant in the room.

**Comment by batiste** [comment]
Also note that from pep08

"Python packages should also have short, all-lowercase names, although the use of underscores is discouraged."

**Comment by remik** [comment]
Hi,
We will try to find some time next week to review changes and do few PR.
Change name could be a problem with migration but we will try to do some command to copy contents.
After this I think we can close this project and contribute to yours.
Sorry for any "bad" feelings but as I mention before we did it because your projects looks dead - no activity, no response for PR (16 May 2012  - 15 Mar 2013)

**Comment by jpic** [comment]
It's cool to fork an un-maintained project, but next time try to keep the git history, it can be useful (license, copyright and authors should be kept too but that I think is clear for everybody now ^^).

If a project need maintainers, non-profit orgs like djangonauts or yourlabs might be able to help.

**Comment by batiste** [comment]
You are talking about those 2 pull requests?

16 May 2012 https://github.com/batiste/django-page-cms/pull/60
15 Mar 2013 https://github.com/batiste/django-page-cms/pull/64

The first one was opened and merged on the same day. It is difficult to be more reactive than that. The second one took me 3 month to merge. Maybe this is too much? Who knows why I took so long. I can't recall.

Anyway we are not here to nitpick, if anybody wants to become part of the maintenance for this project I am all for it.

The migration path for the data is indeed an issue, you have to rename the tables especially with foreign keys, but I think this is possible.

http://stackoverflow.com/questions/25091130/django-migration-strategy-for-renaming-a-model-and-relationship-fields

**Comment by batiste** [comment]
Sorry it was for renaming models, not the entire app. This is a different problem

**Comment by remik** [comment]
I am guessing dates now, I don't remember exactly. I just remember that it looks like dead. Sorry for wrong checking.

I don't remember why we didn't fork it. I did not copy it so I can't say for someone.

Somehow we did migration in past so I need to find our script (TODO: clean, tests, release) and use.

Just repeat: we will be happy to do PR and help with next features. Our users are happy (maybe except import/export but this longer talk) with this package and we already integrating it in all our projects so we want to do it better.

**Comment by batiste** [comment]
I draw your attention towards those commands I wrote:

https://github.com/batiste/django-page-cms/blob/master/pages/management/commands/pages_push.py

The provide a way to pull and push content from one instance of the CMS to another rather simply using the CMS API.

This might prove useful in some way. This is definitely a similar feature that the one provided by the JSON import/export plugin somehow... But more developer friendly.

**Comment by remik** [comment]
Just for tracking what to do:
 - [ ] Grapelli support (this is big as all ArabellaTech clients want to use Grappeli admin theme)
 - [ ] HTML Meta support - few metadata fields on all pages (in admin collapsed by default) 
 - [ ]  update CSS/JS lib to new versions (this might be already outdated)
 - [ ] tox support to easier local testing
 - [ ]  add tags to export/import
 - [ ] New template tag get_pages_with_tag
 - [ ] do redirect if slug is changed - this might need settings to not break old functionality
 - [ ] we had a lot issues with export/import but original project looks like have rewritten this functions so not sure if it is fixed
 - [ ] Check PAGE_TEMPLATES (PR by jpic - https://github.com/ArabellaTech/django-basic-cms/pull/9) - I would ask him to apply patch to keep original author

In our backlog we have few features (early stage or not started):
 - [ ] support files in export/import - this is huge for our clients to easier transfer between stages
 - [ ] support denormalise file names after upload eg. remove spaces as it breaks sometimes front-end code
 - [ ] support for Django 1.9 - I saw you started work - do you need help? A lot tests fails.

**Comment by jpic** [comment]
@remik did you consider django-suit ?

**Comment by remik** [comment]
No, I might check it too. We are using Grappeli for our clients because of extra functions (inline sort, auto apply action, put inlines between formset etc) not style but looks like django-suit is similar.
Also we did Grappeli fixes in this fork.

**Comment by jpic** [comment]
Maybe you can check my item in your list in favor batiste/django-page-cms#122. Keep up the great work B)

**Comment by remik** [comment]
Thanks!

**Comment by remik** [comment]
We are very close. What left before migration script/doc is: Support denormalise file names after upload eg. remove spaces as it breaks sometimes front-end code

Also @jacoor we need to think what to do with API? It is possible other structure. Do you want support both ways in django-basic-cms to give frontend time to transition (before lib will be replaced) or just force them to use new version (when lib will be replaced)

**Comment by jacoor** [comment]
I wouldn't worry too much about FE. We should try the original project on
one of new projects we do and see how it works for us, fix gaps, if any.
FE will survive.

2016-05-31 4:34 GMT+02:00 Remigiusz Dymecki <notifications@github.com>:

> We are very close. What left before migration script/doc is: Support
> denormalise file names after upload eg. remove spaces as it breaks
> sometimes front-end code
>
> Also @jacoor <https://github.com/jacoor> we need to think what to do with
> API? It is possible other structure. Do you want support both ways in
> django-basic-cms to give frontend time to transition (before lib will be
> replaced) or just force them to use new version (when lib will be replaced)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ArabellaTech/django-basic-cms/issues/11#issuecomment-222579868>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA8YSBy7HDxYE4x4SyH-WitSKssVWHFcks5qG54tgaJpZM4GBst7>
> .
>



-- 
Jacek Ostański
Internet Evolution - Strony i serwisy Internetowe
608-230-735

**Comment by remik** [comment]
Last "feature" done. 
Will work on migration soon - not sure when as I need to back to work after conference.

**Comment by jacoor** [comment]
@remik what is the status here?

**Comment by remik** [comment]
On hold.
Missing migration option only.

**Comment by batiste** [comment]
Hello, just wanted to let you know I had time to update the CMS those last 2 days. I fixed up some frontend bugs in the admin and added a script to create new website easily

http://django-page-cms.readthedocs.io/en/latest/changelog.html#release-1-9-4

I changed the CSS for the admin so it might be a good idea to check if everything still works with grapelli.

Lately I have been doing a lot of migration for Django so I might be able to help there.

Cheers,
Batiste
